### PR TITLE
Implement Product's Open Graph

### LINKED ISSUE #3447: Implement Product's Open Graph
Specs (currently not responding): https://developers.facebook.com/docs/reference/opengraph/object-type/product/

This can be helpful as well: https://webcode.tools/open-graph-generator/product-item

All tags in the snippet below should be present on the product page:
```
<meta property="og:type" content="product.item">
<meta property="og:title" content="My red product">
<meta property="og:url" content="https://fullabsoluteurl.com/my-red-product-49/">
<meta property="og:image" content="https://fullabsoluteurl.com/media/products/product.jpg">
<meta property="product:condition" content="new">
<meta property="product:availability" content="in stock">
<meta property="product:price:amount" content="49.99">
<meta property="product:price:currency" content="USD">
<meta property="product:retailer_item_id" content="SKU">
```

If implemented properly, snippets, eg. on slack, should contain price ("cena") and color ("kolor")
screenshot 2018-12-12 at 20 22 00

### PR DESCRIPTION
**Pull request by igoryok-zp**
Fixes #3447

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 1722e7c7215b5a34a1b26bd3bd66eeb0d3722085 (igoryok-zp/saleor @ fix/3447-implement-product-s-open-graph)
Base comparison is 3a6fe51c068aca991331d6eb43317102ed68ed09.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         41	         41	             20

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	              2
```

</p>
</details>

File: templates/product/details.html [hunk 1]
```diff
@@ -37,6 +37,12 @@
   {% else %}
     <meta property="og:image" content="{% static "images/block1.png" %}">
   {% endif %}
+
+  <meta property="product:condition" content="new">
+  <meta property="product:availability" content="{% if product.is_in_stock %}in stock{% else %}out of stock{% endif %}">
+  <meta property="product:price:amount" content="{% if site.settings.display_gross_prices %}{{ availability.price_range.start.gross.amount }}{% else %}{{ availability.price_range.net.gross.amount }}{% endif %}">
+  <meta property="product:price:currency" content="{{ availability.price_range.currency }}">
+  <meta property="product:retailer_item_id" content="SKU">
```
Comment (NyanKiyoshi) [inline templates/product/details.html]: We are missing the SKU here

File: templates/product/details.html [hunk 2]
```diff
@@ -37,6 +37,12 @@
   {% else %}
     <meta property="og:image" content="{% static "images/block1.png" %}">
   {% endif %}
+
+  <meta property="product:condition" content="new">
```
Comment (NyanKiyoshi) [inline templates/product/details.html]: Don't we need to define `og:type` as well?
```html
<meta property="og:type" content="product.item">
```

File: templates/product/details.html [hunk 3]
```diff
@@ -37,6 +37,12 @@
...
+  <meta property="product:availability" content="{% if product.is_in_stock %}in stock{% else %}out of stock{% endif %}">
+  <meta property="product:price:amount" content="{% if site.settings.display_gross_prices %}{{ availability.price_range.start.gross.amount }}{% else %}{{ availability.price_range.net.gross.amount }}{% endif %}">
```
Comment (NyanKiyoshi) [inline templates/product/details.html]:
```suggestion
  <meta property="product:price:amount" content="{% if site.settings.display_gross_prices %}{{ availability.price_range.start.gross.amount }}{% else %}{{ availability.price_range.start.net.amount }}{% endif %}">
```

File: templates/product/details.html [hunk 4]
```diff
@@ -37,6 +37,12 @@
...
+  <meta property="product:price:currency" content="{{ availability.price_range.currency }}">
```
Comment (NyanKiyoshi) [inline templates/product/details.html]:
```suggestion
  <meta property="product:price:currency" content="{{ availability.price_range.start.net.currency }}">
```

**Comment by igoryok-zp** [comment]
Hi @NyanKiyoshi, thanks for the review.
I made a mistake with net amount.
`MoneyRange` doesn't define currency, but `availability.price_range.currency` contains value.
I fixed `og:type` and `retailer_item_id`.
Take a look at my updates, pease.

**Comment by maarcingebala** [comment]
@igoryok-zp CI is failing here so I can't merge this PR. Will you have time to fix that?

**Pull request closed without merging**